io/io/v7/src/TFile.cxx

```diff
@@ -191,7 +191,7 @@ std::string ROOT::Experimental::TFile::SetCacheDir(std::string_view path) {
   std::lock_guard<std::mutex> lock(GetCacheDirMutex());
 
   std::string ret = ::TFile::GetCacheFileDir();
-  ::TFile::SetCacheFileDir(path.to_string().c_str());
+  ::TFile::SetCacheFileDir(path.data());
```

Comment:
Don't we need a null terminated string here?
Alternatively, what about adding a string_view overload of ::TFile::SetCacheFileDir ?